glotaran/plugin_system/test/test_base_registry.py

```diff
@@ -45,11 +46,35 @@ def some_other_method(self):
         pass
 
 
-class MockPluginSubclass(MockPlugin):
+class MockPluginSubclassPartial(MockPlugin):
     def some_method(self):
         return "different implementation"
 
 
+class MockPluginSubclassFull(MockPlugin):
+    def some_method(self):
+        return "different implementation"
+
+    def some_other_method(self):
+        return "different implementation"
+
+
+class MockPluginSubclassStr(MockPlugin):
+    def some_method(self):
+        return "different implementation"
+
+
+def get_mock_plugin_function(plugin_registry_key: str):
+    if plugin_registry_key == "base":
+        return MockPlugin
+    elif plugin_registry_key == "sub_class_partial":
+        return MockPluginSubclassPartial
+    elif plugin_registry_key == "sub_class_full":
+        return MockPluginSubclassFull
+    elif plugin_registry_key == "sub_class_str":
+        return MockPluginSubclassStr
+
```

Comment:
I know it's just a mock function used in testing but maybe fail with not implemented error when called with unsupported key?

Reply:
Fair enough fixed it in [d03982f](https://github.com/glotaran/pyglotaran/pull/1229/commits/d03982f090549ae82893eee5965249c7c27a32ac).